cmd/find.go

```diff
@@ -0,0 +1,493 @@
+ * limitations under the License.
+ */
+
+// Package cmd stores all of the mc utilities
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+
+	// golang does not support flat keys for path matching, find does
+	"github.com/minio/minio/pkg/wildcard"
+)
+
+// findMSG holds JSON and string values for printing
+type findMSG struct {
+	Path string `json:"path"`
+}
+
+// String calls tells the console what to print and how to print it
+func (f findMSG) String() string {
+	return console.Colorize("Find", f.Path)
+}
+
+// JSON formats output to be JSON output
+func (f findMSG) JSON() string {
+	f.Path = "path"
+	jsonMessageBytes, e := json.Marshal(f)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// nameMatch pattern matches off of the base of the filepath
+func nameMatch(path, pattern string) (bool, error) {
+	base := filepath.Base(path)
+
+	return filepath.Match(pattern, base)
+}
+
+// pathMatch pattern matches off of of the entire filepath
+func pathMatch(path, pattern string) bool {
+	return wildcard.Match(pattern, path)
+}
+
+// regexMatch pattern matches off of the entire filepath using regex library
+func regexMatch(path, pattern string) (bool, error) {
+	return regexp.MatchString(pattern, path)
+}
+
+// doFindPrint prints the output in accordance with the supplied substitution arguments
+func doFindPrint(path string, ctx *cli.Context, fileContent contentMessage) {
+	printString := SubArgsHelper(ctx.String("print"), path, fileContent)
+	printMsg(findMSG{
+		Path: printString,
+	})
+}
+
+// doFindExec passes the users input along to the command line, also dealing with substitution arguments
+func doFindExec(ctx *cli.Context, path string, fileContent contentMessage) {
+	commandString := SubArgsHelper(ctx.String("exec"), path, fileContent)
+	commandArgs := strings.Split(commandString, " ")
+
+	cmd := exec.Command(commandArgs[0], commandArgs[1:]...)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		console.Fatalln(err)
+		console.Fatalln()
+	}
+	console.Println(string(out.Bytes()))
+}
+
+// doFindWatch watches the server side to see if a given action is preformed, if yes then a can provided by exec can be executed
+func doFindWatch(path string, ctx *cli.Context) {
+	params := watchParams{
+		recursive: true,
+		accountID: fmt.Sprintf("%d", time.Now().Unix()),
+		events:    []string{"put"},
+	}
+
+	pathnameParts := strings.SplitAfter(path, "/")
+	alias := strings.TrimSuffix(pathnameParts[0], "/")
+
+	_, e := getHostConfig(alias) // extract the hostname alias from the path name if present
+	noAlias := (e == nil)
+
+	clnt, content, err := url2Stat(path)
+	errorIf(err, "Unable to construct client")
```

Comment:
Could you check https://github.com/minio/mc/pull/2201#pullrequestreview-53376254 and make the changes?